conans/client/generators/__init__.py

```diff
@@ -130,12 +130,15 @@ def deactivates(filenames):
         bats = []
         shs = []
         ps1s = []
+        fishes = []
         for env_script in env_scripts:
             path = os.path.join(conanfile.generators_folder, env_script)
             if env_script.endswith(".bat"):
                 bats.append(path)
             elif env_script.endswith(".sh"):
                 shs.append(subsystem_path(subsystem, path))
+            elif env_script.endswith(".fish"):
+                fishes.append(subsystem_path(subsystem, path))
```

Comment:
fish exists/work in msys2 and other subsystems?

Reply:
Yes! I also didn't think it would, but I was pleasently supprised that a package for msys2 exists and so I added the subsystem path call here once I found out :)